src/freenet/clients/http/SecurityLevelsToadlet.java

```diff
@@ -168,32 +170,32 @@ public void handleMethodPOST(URI uri, HTTPRequest request, ToadletContext ctx) t
 								addBackToSeclevelsLink(content);
 
 								writeHTMLReply(ctx, 200, "OK", pageNode.generate());
-								if(changedAnything)
+								if (changedAnything)
 									core.storeConfig();
 								return;
 							} catch (MasterKeysFileSizeException e) {
 								sendPasswordFileCorruptedPage(e.isTooBig(), ctx, false, true);
-								if(changedAnything)
+								if (changedAnything)
 									core.storeConfig();
 								return;
 							}
 						} else {
-							sendPasswordPage(ctx, password != null && password.length() == 0, newPhysicalLevel.name());
-							if(changedAnything)
+							sendPasswordPage(ctx, password.length() == 0 && confirmPassword.length() == 0, newPhysicalLevel.name());
+							if (changedAnything)
 								core.storeConfig();
 							return;
 						}
 					}
-					if((newPhysicalLevel == PHYSICAL_THREAT_LEVEL.LOW || newPhysicalLevel == PHYSICAL_THREAT_LEVEL.NORMAL) &&
+					if ((newPhysicalLevel == PHYSICAL_THREAT_LEVEL.LOW || newPhysicalLevel == PHYSICAL_THREAT_LEVEL.NORMAL) &&
 							oldPhysicalLevel == PHYSICAL_THREAT_LEVEL.HIGH) {
 						// Check for password
 						String password = request.getPartAsStringFailsafe("masterPassword", SecurityLevelsToadlet.MAX_PASSWORD_LENGTH);
-						if(password != null && password.length() > 0) {
+						if (password != null && password.length() > 0) {
```

Comment:
This was one of the style changes that I should have put into another commit, I have removed the unneeded null check though.